components/new-contribution-flow/StepPayment.js

```diff
@@ -1,16 +1,264 @@
+import styled from 'styled-components';
 
-import { Box } from '../../components/Grid';
+import { API_V2_CONTEXT, gqlV2 } from '../../lib/graphql/helpers';
 
-class NewContributionFlowStepPayment extends React.Component {
-  static propTypes = {
-    collective: PropTypes.object,
-  };
+import Container from '../../components/Container';
+import { Box, Flex } from '../../components/Grid';
+import Link from '../../components/Link';
+import Loading from '../../components/Loading';
+import MessageBox from '../../components/MessageBox';
+import NewCreditCardForm from '../../components/NewCreditCardForm';
+import { withStripeLoader } from '../../components/StripeProvider';
+import StyledRadioList from '../../components/StyledRadioList';
+import { P } from '../../components/Text';
+import { withUser } from '../../components/UserProvider';
+
+import { ERROR_MESSAGES, PAYMENT_METHOD_TYPES } from './constants';
+import { generatePaymentMethodOptions } from './utils';
 
-  render() {
-    return <Box width={1}>Payment method chooser - same for all states</Box>;
+const PaymentMethodBox = styled(Container)`
+  border-top: 1px solid ${themeGet('colors.black.300')};
+  display: flex;
+  flex-direction: column;
+  background: ${themeGet('colors.white.full')};
+  &:hover {
+    background: ${themeGet('colors.black.50')};
   }
-}
+`;
+
+const paymentMethodsQuery = gqlV2/* GraphQL */ `
+  query ContributionFlowPaymentMethods($slug: String) {
+    account(slug: $slug) {
+      id
+      paymentMethods(types: ["creditcard", "virtualcard", "prepaid", "collective"]) {
+        id
+        name
+        data
+        service
+        type
+        balance {
+          value
+          currency
+        }
+        account {
+          id
+        }
+        limitedToCollectiveIds
+        limitedToHostCollectiveIds
+      }
+    }
+  }
+`;
+
+const messages = defineMessages({
+  [ERROR_MESSAGES.ERROR_LOW_BALANCE]: {
+    id: 'NewContribute.noCollectiveBalance',
+    defaultMessage:
+      'The balance of this collective is too low to make orders from it. Add funds to {collective} by making a donation to it first.',
+  },
+  [ERROR_MESSAGES.ERROR_DIFFERENT_HOST]: {
+    id: 'NewContribute.noCollectivePaymentMethodsAvailable.differentHost',
+    defaultMessage: 'You cannot make donations to the a Collective that has a different fiscal Host.',
+  },
+  [ERROR_MESSAGES.ERROR_NO_PAYMENT_METHODS]: {
+    id: 'NewContribute.noPaymentMethodsAvailable',
+    defaultMessage: 'There are no payment methods available.',
+  },
+});
+
+const NewContributionFlowStepPayment = ({
+  LoggedInUser,
+  stepDetails,
+  stepProfile,
+  stepPayment,
+  collective,
+  onChange,
+  loadStripe,
+  hideCreditCardPostalCode,
+}) => {
+  const intl = useIntl();
+  const [loadingSelectedPaymentMethod, setLoadingSelectedPaymentMethod] = useState(true);
+  const [paymentMethodsError, setPaymentMethodsError] = useState(false);
+  // eslint-disable-next-line no-unused-vars
```

Comment:
Better to not merge this if possible because there's always a risk of forgetting it. Can't we do this instead?


```es6
const [_, setStripe] = useState(null);
```

Reply:
I actually didn't realise you could do that! Awesome 👍